rep-2017.rst

```diff
@@ -0,0 +1,359 @@
+REP: 2017
+Title: Thread attributes configuration support in rcl
+Author: Masaaki Ueno, Koichi Oyama, Shoji Morita <s-morita@esol.co.jp>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 11-Jul-2023
+Post-History: 18-Aug-2023
+
+
+Abstract
+========
+
+This REP proposed a feature of the ros client library (rcl) to accept the thread attributes, such as priorities, scheduling, and core affinity, which are to be passed to the language bindings like rclcpp.
```

Comment:
@razr 
> Do you extend ROS Utils or ROS Client Library?
 Yes. The prototype implementations have been shared as the pull requests below.
https://github.com/ros2/rcl/pull/1075
https://github.com/ros2/rcutils/pull/424

 > Does it also allow to modify a thread name?

If "it" refers to executors in the language bindings (e.g., rclcpp), yes, they can use parameters provided through command line arguments or environment variables to set their own thread names.

> Is CPU Affinity POSIX compliant?

CPU core affinity is not POSIX compliant but is prevalent in most POSIX-like environments.

> Is it a goal to stay POSIX compliant?

Not strictly, but we might as well refer to the specification to decide the practical parameter set.

> Do you support Windows as well?

This proposal is not specific to any environment because the extension is just to receive the thread parameters at the rcl layer. But, if you want to use this infrastructure from above the language bindings (like rclcpp), you will need to implement some environment-specific code or modules to use those thread attributes, as I proposed for rcpputils in the pull request below.
https://github.com/ros2/rcpputils/pull/179
In the above pull request, I proposed the rclcpp::thread class as an extension of C++ std::thread. Currently, the extension is only applicable to Linux environments. And, for other environments (like Windows or Mac environments), the rclcpp::thread acts like existing std::thread. But, you can extend the rclcpp::thread to use the thread attributes settings for Windows or Mac environments by adding some implementations to rclcpp::thread.

Reply:
> I'm not sure whether `accept` is the right word here. 

I've replaced it with "receive" in the commit below.
https://github.com/ros-infrastructure/rep/pull/385/commits/c06a453f98c06602bd4b6c55d0150f9154ad07ac